### PR TITLE
fix: int64 conversion for collector_id in datasource + add docs

### PR DESCRIPTION
**Pull request by brewkode**
This PR fixed int64 conversion related issue of collector_id field in the http_source datasource.

I started using this datasource and was running into an issue where usage of the http_source data block would cause panic and result int crash log. The relevant stack trace is below. I noticed that there was a conversion issue. I also noticed that the existing data source file already had one round of change which was geared towards this #177. however, it looks like the conversion strategy applied there does not fix it properly.  This fix here is to really address what #177 tried to do.

I've also added a minimal datasource doc to address #242 (which I raised a few days ago). 

```
2021-07-22T23:29:40.920-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: panic: interface conversion: interface {} is int, not int64
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: goroutine 43 [running]:
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: github.com/SumoLogic/terraform-provider-sumologic/sumologic.dataSourceSumologicHTTPSourceRead(0xc00047c380, 0x1a91ce0, 0xc00007f590, 0xc00047c380, 0x0)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        github.com/SumoLogic/terraform-provider-sumologic/sumologic/data_source_sumologic_http_source.go:58 +0x5b3
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).ReadDataApply(0xc0000ef400, 0xc0006ad020, 0x1a91ce0, 0xc00007f590, 0xc0004231b0, 0x1, 0x0)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        github.com/hashicorp/terraform-plugin-sdk@v1.7.0/helper/schema/resource.go:398 +0x91
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).ReadDataApply(0xc0000ef600, 0xc00049dad8, 0xc0006ad020, 0xc0006ad020, 0x0, 0x0)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        github.com/hashicorp/terraform-plugin-sdk@v1.7.0/helper/schema/provider.go:451 +0x8f
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc00000e0d8, 0x1c1e130, 0xc000181800, 0xc000502bc0, 0xc00000e0d8, 0xc000181800, 0xc0002e4b80)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        github.com/hashicorp/terraform-plugin-sdk@v1.7.0/internal/helper/plugin/grpc_provider.go:1036 +0x42d
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadDataSource_Handler(0x1a61300, 0xc00000e0d8, 0x1c1e130, 0xc000181800, 0xc00056c660, 0x0, 0x1c1e130, 0xc000181800, 0xc00054aa20, 0x84)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        github.com/hashicorp/terraform-plugin-sdk@v1.7.0/internal/tfplugin5/tfplugin5.pb.go:3225 +0x214
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000093080, 0x1c257d8, 0xc000131500, 0xc00058a200, 0xc000469b30, 0x20ea070, 0x0, 0x0, 0x0)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        google.golang.org/grpc@v1.27.1/server.go:1024 +0x522
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: google.golang.org/grpc.(*Server).handleStream(0xc000093080, 0x1c257d8, 0xc000131500, 0xc00058a200, 0x0)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        google.golang.org/grpc@v1.27.1/server.go:1313 +0xd2c
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc0000b41e0, 0xc000093080, 0x1c257d8, 0xc000131500, 0xc00058a200)
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        google.golang.org/grpc@v1.27.1/server.go:722 +0xab
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7: created by google.golang.org/grpc.(*Server).serveStreams.func1
2021-07-22T23:29:40.921-0700 [DEBUG] plugin.terraform-provider-sumologic_v2.9.7:        google.golang.org/grpc@v1.27.1/server.go:720 +0xa5
```

### CODE REVIEW

**Comment by brewkode** [comment]
@sumovishal @seansain @maimaisie @vsinghal13 -- Can one you please take a look at this small change ?

File: website/docs/d/http_source.html.markdown
Comment (vsinghal13) [inline website/docs/d/http_source.html.markdown:22]: @frankreno should we make these fields mandatory?

**Review by vsinghal13 — COMMENTED** [review]
Thanks @brewkode for the PR. The int64 change LGTM. 

Could you separate out the docs change to another PR so that we can discuss it separately and then decide whether we should change the data source parameters to be mandatory to avoid failure with a panic?

File: sumologic/data_source_sumologic_http_source.go
Comment (pmalek-sumo) [inline sumologic/data_source_sumologic_http_source.go:58]: I'd actually add a check flag `ok` (type conversion second param) and check for both `int` and `int64` in a type switch.

The reason for this is as I understand this, is that this will be an `int` on 64 bit platforms and `int64` on different platforms (e.g. 32bit).

**Comment by brewkode** [comment]
> Thanks @brewkode for the PR. The int64 change LGTM.
> 
> Could you separate out the docs change to another PR so that we can discuss it separately and then decide whether we should change the data source parameters to be mandatory to avoid failure with a panic?

I pushed in that change now. Separate docs MR here #251

File: sumologic/data_source_sumologic_http_source.go
Comment (brewkode) [reply sumologic/data_source_sumologic_http_source.go:58]: @pmalek-sumo Do you intend to do something like this ?


```go
d64, ok64 := d.Get("collector_id").(int64)
var collector_id int64
if ok64 {
  collector_id = d64
} else {
  dint, ok = d.Get("collector_id").(int)
  if !ok {
     return error
  }
  collector_id = int64(dint)
}
```

Comment (pmalek-sumo) [reply sumologic/data_source_sumologic_http_source.go:58]: Pretty much but using a `switch` and a type switch:

```go
...
	var collectorId int64
	switch v := d.Get("collector_id").(type) {
	case int:
		collectorId = int64(v)
	case int64:
		collectorId = v
	default:
		return fmt.Errorf("unknown data type of collector_id: %T, value: %v", v, v)
	}

	source, err := c.GetSourceName(collectorId, d.Get("name").(string))
	if err != nil {
		return err
	}
...
```

**Comment by brewkode** [comment]
@pmalek-sumo PTAL latest change.

**Review by pmalek-sumo — APPROVED** [review]

**Comment by brewkode** [comment]
@vsinghal13 @maimaisie @seansain Please take a look at the change and share your reviews.

**Review by vsinghal13 — APPROVED** [review]

**Comment by brewkode** [comment]
@sumovishal @maimaisie @seansain Can you all please review this change ?

**Comment by sumovishal** [comment]
> @sumovishal @maimaisie @seansain Can you all please review this change ?

@brewkode you can go ahead and merge it. @vsinghal13 has given his blessings.

**Comment by brewkode** [comment]
@sumovishal I don't have write access to this repo. So, I cannot merge this. Can you please help merge this ?

**Comment by sumovishal** [comment]
> @sumovishal I don't have write access to this repo. So, I cannot merge this. Can you please help merge this ?

yeah, I just realized that. Don't worry, we will take care of it.

**Comment by brewkode** [comment]
@sumovishal I've a hard dependency on this. Do you have some idea of when your team will be able to get to this ?

**Comment by sumovishal** [comment]
@brewkode we will do a new release of the provider by Friday.

**Comment by sumovishal** [comment]
> @brewkode we will do a new release of the provider by Friday.

https://github.com/SumoLogic/terraform-provider-sumologic/releases/tag/v2.9.8

**Pull request merged**
